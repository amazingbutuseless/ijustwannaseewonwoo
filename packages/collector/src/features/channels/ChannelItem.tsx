import React from 'react';

import { ChannelItemThumbnails } from '../../types';

import { ChannelButton } from './ChannelItem.style';

export interface ChannelItemProps {
  id: string;
  title: string;
  thumbnails: ChannelItemThumbnails;
  onClick?: (videoId: string) => void;
}

export default function ChannelItem({ id, title, thumbnails, onClick }: ChannelItemProps) {
  const handleClick = () => {
    onClick(id);
  };

  return (
    <li>
      <ChannelButton
        type="button"
        thumbnails={thumbnails}
        data-id={id}
        aria-label={title}
        onClick={handleClick}
      >
        <span>{title}</span>
      </ChannelButton>
    </li>
  );
}
