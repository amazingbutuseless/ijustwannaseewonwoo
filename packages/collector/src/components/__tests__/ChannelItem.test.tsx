import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ChannelItem from '../ChannelItem';

const onClick = jest.fn();

function render() {
  const thumbnails = {
    default: 'thumbnail-default.png',
    high: 'thumbnail-high.png',
  };

  return shallow(
    <ChannelItem id="testid" title="Test Channel" thumbnails={thumbnails} onClick={onClick} />
  );
}

beforeEach(() => {
  onClick.mockClear();
});

describe('ChannelItem', () => {
  it('must be rendered', () => {
    const item = render();

    expect(toJson(item)).toMatchSnapshot();
  });

  it('onClick fires when the item is clicked', () => {
    const item = render();

    item.simulate('click');
    expect(onClick).toBeCalledWith('testid');
  });
});
