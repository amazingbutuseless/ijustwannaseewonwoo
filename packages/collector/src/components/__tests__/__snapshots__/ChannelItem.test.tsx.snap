// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelItem must be rendered 1`] = `
<Styled(button)
  aria-label="Test Channel"
  data-id="testid"
  onClick={[Function]}
  thumbnails={
    Object {
      "default": "thumbnail-default.png",
      "high": "thumbnail-high.png",
    }
  }
  type="button"
>
  <span>
    Test Channel
  </span>
</Styled(button)>
`;
